### DISCUSSION TITLE
usage of ConvexClippingPolygon2d class obj

### DISCUSSION THREAD

**Discussion by Ganti4270** [mozman/ezdxf#1142] [Q&A]
hi I am new to ezdxf, firstly it is a great packaage thank you for makeing it OpenSource I have a question during my exploring of the package features I encountred a class ConvexClippingPolygon2d in the clipping.py, i thought that may be helpful to my requirements and  i tried to get the hang of it by this code # Define vertices of the convex polygon
vertex_tuples  = [(0, 0), (4, 0), (4, 4), (0, 4)]
vertices = [Vec2(x, y) for x, y in vertex_tuples]
#print(vertices)
# Create a ConvexClippingPolygon2d object
polygon = ConvexClippingPolygon2d(vertices) and I get this error

    if clip[0].isclose(clip[-1], abs_tol=self.abs_tol):

AttributeError: 'tuple' object has no attribute 'isclose'
in the clipping.py  

am i missing something ? please help

**Comment by cococosusu** [comment]
Hi !
Which ezdxf version do you use?
I'm using version 1.3.2, and the code you wrote works well!

**Reply by Ganti4270** [reply]
version 1.3.2 
I installed the ezdxf few days back

**Comment by mozman** [comment]
```Python
from ezdxf.math import Vec2

if Vec2(clip[0]).isclose(clip[-1], abs_tol=self.abs_tol):
```
